### DISCUSSION TITLE
Error when installing the extension

### DISCUSSION THREAD

**Discussion by GrandaddyShmax** [thygate/stable-diffusion-webui-depthmap-script#91] [Q&A] [answered]
It throws me this error when i try to install it through Automatic1111 WebUi

stderr: ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'V:\\Stable Diffussion Webui\\stable-diffusion-webui\\venv\\Lib\\site-packages\\~il\\_imaging.cp310-win_amd64.pyd'
Check the permissions.

**Comment by GrandaddyShmax** [comment]
This is the full error if needed:

Error running install.py for extension V:\Stable Diffussion Webui\stable-diffusion-webui\extensions\stable-diffusion-webui-depthmap-script.
Command: "V:\Stable Diffussion Webui\stable-diffusion-webui\venv\Scripts\python.exe" "V:\Stable Diffussion Webui\stable-diffusion-webui\extensions\stable-diffusion-webui-depthmap-script\install.py"
Error code: 1
stdout: Installing vispy requirement for depthmap script
Installing rembg requirement for depthmap script

stderr: Traceback (most recent call last):
  File "V:\Stable Diffussion Webui\stable-diffusion-webui\extensions\stable-diffusion-webui-depthmap-script\install.py", line 15, in <module>
    launch.run_pip("install rembg", "rembg requirement for depthmap script")
  File "V:\Stable Diffussion Webui\stable-diffusion-webui\launch.py", line 93, in run_pip
    return run(f'"{python}" -m pip {args} --prefer-binary{index_url_line}', desc=f"Installing {desc}", errdesc=f"Couldn't install {desc}")
  File "V:\Stable Diffussion Webui\stable-diffusion-webui\launch.py", line 64, in run
    raise RuntimeError(message)
RuntimeError: Couldn't install rembg requirement for depthmap script.
Command: "V:\Stable Diffussion Webui\stable-diffusion-webui\venv\Scripts\python.exe" -m pip install rembg --prefer-binary
Error code: 1
stdout: Collecting rembg
  Downloading rembg-2.0.30-py3-none-any.whl (13 kB)
Collecting fastapi~=0.87.0
  Downloading fastapi-0.87.0-py3-none-any.whl (55 kB)
     -------------------------------------- 55.5/55.5 kB 718.5 kB/s eta 0:00:00
Collecting pillow~=9.3.0
  Downloading Pillow-9.3.0-cp310-cp310-win_amd64.whl (2.5 MB)
     ---------------------------------------- 2.5/2.5 MB 6.8 MB/s eta 0:00:00
Collecting imagehash~=4.3.1
  Downloading ImageHash-4.3.1-py2.py3-none-any.whl (296 kB)
     ---------------------------------------- 296.5/296.5 kB ? eta 0:00:00
Collecting python-multipart~=0.0.5
  Downloading python-multipart-0.0.5.tar.gz (32 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting numpy~=1.23.5
  Downloading numpy-1.23.5-cp310-cp310-win_amd64.whl (14.6 MB)
     --------------------------------------- 14.6/14.6 MB 27.3 MB/s eta 0:00:00
Collecting opencv-python-headless~=4.6.0.66
  Downloading opencv_python_headless-4.6.0.66-cp36-abi3-win_amd64.whl (35.5 MB)
     --------------------------------------- 35.5/35.5 MB 28.4 MB/s eta 0:00:00
Collecting pymatting~=1.1.8
  Downloading PyMatting-1.1.8-py3-none-any.whl (47 kB)
     ---------------------------------------- 47.7/47.7 kB ? eta 0:00:00
Collecting scikit-image~=0.19.3
  Using cached scikit_image-0.19.3-cp310-cp310-win_amd64.whl (12.0 MB)
Requirement already satisfied: tqdm~=4.64.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from rembg) (4.64.1)
Collecting watchdog~=2.1.9
  Downloading watchdog-2.1.9-py3-none-win_amd64.whl (78 kB)
     ---------------------------------------- 78.4/78.4 kB ? eta 0:00:00
Requirement already satisfied: click~=8.1.3 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from rembg) (8.1.3)
Collecting scipy~=1.9.3
  Downloading scipy-1.9.3-cp310-cp310-win_amd64.whl (40.1 MB)
     --------------------------------------- 40.1/40.1 MB 28.5 MB/s eta 0:00:00
Collecting asyncer~=0.0.2
  Downloading asyncer-0.0.2-py3-none-any.whl (8.3 kB)
Collecting onnxruntime~=1.13.1
  Downloading onnxruntime-1.13.1-cp310-cp310-win_amd64.whl (5.9 MB)
     ---------------------------------------- 5.9/5.9 MB 31.6 MB/s eta 0:00:00
Requirement already satisfied: aiohttp~=3.8.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from rembg) (3.8.3)
Collecting uvicorn~=0.20.0
  Downloading uvicorn-0.20.0-py3-none-any.whl (56 kB)
     ---------------------------------------- 56.9/56.9 kB ? eta 0:00:00
Collecting pooch~=1.6.0
  Downloading pooch-1.6.0-py3-none-any.whl (56 kB)
     ---------------------------------------- 56.3/56.3 kB ? eta 0:00:00
Collecting filetype~=1.2.0
  Downloading filetype-1.2.0-py2.py3-none-any.whl (19 kB)
Requirement already satisfied: yarl<2.0,>=1.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (1.8.1)
Requirement already satisfied: multidict<7.0,>=4.5 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (6.0.2)
Requirement already satisfied: frozenlist>=1.1.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (1.3.1)
Requirement already satisfied: charset-normalizer<3.0,>=2.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (2.1.1)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (4.0.2)
Requirement already satisfied: attrs>=17.3.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (22.1.0)
Requirement already satisfied: aiosignal>=1.1.2 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from aiohttp~=3.8.1->rembg) (1.2.0)
Requirement already satisfied: anyio<4.0.0,>=3.4.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from asyncer~=0.0.2->rembg) (3.6.1)
Requirement already satisfied: colorama in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from click~=8.1.3->rembg) (0.4.5)
Collecting starlette==0.21.0
  Downloading starlette-0.21.0-py3-none-any.whl (64 kB)
     ---------------------------------------- 64.0/64.0 kB ? eta 0:00:00
Requirement already satisfied: pydantic!=1.7,!=1.7.1,!=1.7.2,!=1.7.3,!=1.8,!=1.8.1,<2.0.0,>=1.6.2 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from fastapi~=0.87.0->rembg) (1.10.2)
Requirement already satisfied: PyWavelets in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from imagehash~=4.3.1->rembg) (1.4.1)
Requirement already satisfied: packaging in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime~=1.13.1->rembg) (21.3)
Collecting coloredlogs
  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
     ---------------------------------------- 46.0/46.0 kB ? eta 0:00:00
Requirement already satisfied: protobuf in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime~=1.13.1->rembg) (3.19.6)
Collecting sympy
  Downloading sympy-1.11.1-py3-none-any.whl (6.5 MB)
     ---------------------------------------- 6.5/6.5 MB 34.5 MB/s eta 0:00:00
Requirement already satisfied: flatbuffers in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime~=1.13.1->rembg) (22.9.24)
Collecting appdirs>=1.3.0
  Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Requirement already satisfied: requests>=2.19.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from pooch~=1.6.0->rembg) (2.25.1)
Requirement already satisfied: numba!=0.49.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from pymatting~=1.1.8->rembg) (0.56.2)
Requirement already satisfied: six>=1.4.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from python-multipart~=0.0.5->rembg) (1.16.0)
Requirement already satisfied: tifffile>=2019.7.26 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from scikit-image~=0.19.3->rembg) (2022.8.12)
Requirement already satisfied: imageio>=2.4.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from scikit-image~=0.19.3->rembg) (2.22.0)
Requirement already satisfied: networkx>=2.2 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from scikit-image~=0.19.3->rembg) (2.8.6)
Requirement already satisfied: h11>=0.8 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from uvicorn~=0.20.0->rembg) (0.12.0)
Requirement already satisfied: idna>=2.8 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from anyio<4.0.0,>=3.4.0->asyncer~=0.0.2->rembg) (2.10)
Requirement already satisfied: sniffio>=1.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from anyio<4.0.0,>=3.4.0->asyncer~=0.0.2->rembg) (1.3.0)
Requirement already satisfied: setuptools<60 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from numba!=0.49.0->pymatting~=1.1.8->rembg) (59.8.0)
Requirement already satisfied: llvmlite<0.40,>=0.39.0dev0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from numba!=0.49.0->pymatting~=1.1.8->rembg) (0.39.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from packaging->onnxruntime~=1.13.1->rembg) (3.0.9)
Requirement already satisfied: typing-extensions>=4.1.0 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from pydantic!=1.7,!=1.7.1,!=1.7.2,!=1.7.3,!=1.8,!=1.8.1,<2.0.0,>=1.6.2->fastapi~=0.87.0->rembg) (4.3.0)
Requirement already satisfied: chardet<5,>=3.0.2 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from requests>=2.19.0->pooch~=1.6.0->rembg) (4.0.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from requests>=2.19.0->pooch~=1.6.0->rembg) (1.26.12)
Requirement already satisfied: certifi>=2017.4.17 in v:\stable diffussion webui\stable-diffusion-webui\venv\lib\site-packages (from requests>=2.19.0->pooch~=1.6.0->rembg) (2022.9.24)
Collecting humanfriendly>=9.1
  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
     ---------------------------------------- 86.8/86.8 kB ? eta 0:00:00
Collecting mpmath>=0.19
  Downloading mpmath-1.2.1-py3-none-any.whl (532 kB)
     ------------------------------------- 532.6/532.6 kB 32.6 MB/s eta 0:00:00
Collecting pyreadline3
  Downloading pyreadline3-3.4.1-py3-none-any.whl (95 kB)
     ---------------------------------------- 95.2/95.2 kB ? eta 0:00:00
Building wheels for collected packages: python-multipart
  Building wheel for python-multipart (setup.py): started
  Building wheel for python-multipart (setup.py): finished with status 'done'
  Created wheel for python-multipart: filename=python_multipart-0.0.5-py3-none-any.whl size=31678 sha256=ca385122f9716c911e6ab3cde7307dce46928a7df73f2d3dd9c4ba72ec54c0ec
  Stored in directory: c:\users\simur\appdata\local\pip\cache\wheels\ae\3f\03\fa4bd98cd7f4a25e63b6a0b61a7a8352e0f874cd9de1f3390d
Successfully built python-multipart
Installing collected packages: pyreadline3, mpmath, filetype, appdirs, watchdog, sympy, python-multipart, pillow, numpy, humanfriendly, uvicorn, starlette, scipy, pooch, opencv-python-headless, coloredlogs, asyncer, scikit-image, pymatting, onnxruntime, imagehash, fastapi, rembg
  Attempting uninstall: python-multipart
    Found existing installation: python-multipart 0.0.4
    Uninstalling python-multipart-0.0.4:
      Successfully uninstalled python-multipart-0.0.4
  Attempting uninstall: pillow
    Found existing installation: Pillow 9.4.0
    Uninstalling Pillow-9.4.0:
      Successfully uninstalled Pillow-9.4.0

stderr: ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'V:\\Stable Diffussion Webui\\stable-diffusion-webui\\venv\\Lib\\site-packages\\~il\\_imaging.cp310-win_amd64.pyd'
Check the permissions.

**Comment by graemeniedermayer** [comment, marked as answer]
This looks very similar to issue #89 they suggested deleting the config file and restarting helped. 

I think this error might stem from rembg having `pillow==9.3.0` in it's requirement.txt file and pip trying to uninstall pillow while it's in use.

**Reply by GrandaddyShmax** [reply]
you mean config.json that is in the main folder?

**Reply by graemeniedermayer** [reply]
Yah auto1111 will remake a new config.json file when you start it up again (renaming config.json should work too).
